### PR TITLE
Add table aws_elastic_beanstalk_environment. closes #157

### LINKED ISSUE #157: Add table aws_elastic_beanstalk_environment
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/elasticbeanstalk

### PR DESCRIPTION
**Pull request by Priyanka585464**
```
SETUP: tests/aws_elastic_beanstalk_environment []

PRETEST: tests/aws_elastic_beanstalk_environment

TEST: tests/aws_elastic_beanstalk_environment
Running terraform
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_region.alternate: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_iam_role.role: Creating...
aws_vpc.my_vpc: Creating...
aws_iam_service_linked_role.elasticbeanstalk: Creating...
aws_iam_service_linked_role.elasticbeanstalk: Creation complete after 3s [id=arn:aws:iam::986325076436:role/aws-service-role/elasticbeanstalk.amazonaws.com/AWSServiceRoleForElasticBeanstalk]
aws_elastic_beanstalk_application.application_test: Creating...
aws_iam_role.role: Creation complete after 5s [id=test_role]
aws_iam_instance_profile.instance_profile: Creating...
aws_elastic_beanstalk_application.application_test: Creation complete after 5s [id=turbottest67900]
aws_iam_instance_profile.instance_profile: Creation complete after 4s [id=turbottest67900]
aws_vpc.my_vpc: Still creating... [10s elapsed]
aws_vpc.my_vpc: Creation complete after 13s [id=vpc-035564ce57deffb26]
aws_internet_gateway.testinternetgateway: Creating...
aws_subnet.my_subnet1: Creating...
aws_security_group.ssh-allowed: Creating...
aws_internet_gateway.testinternetgateway: Creation complete after 7s [id=igw-0a33e6b72aa338705]
aws_route_table.route: Creating...
aws_subnet.my_subnet1: Still creating... [10s elapsed]
aws_security_group.ssh-allowed: Still creating... [10s elapsed]
aws_route_table.route: Creation complete after 5s [id=rtb-0ddef4b3436efe3d1]
aws_security_group.ssh-allowed: Creation complete after 12s [id=sg-09aba85376ebedb3e]
aws_subnet.my_subnet1: Creation complete after 17s [id=subnet-0717f377ea3c80136]
aws_route_table_association.routeassociation1: Creating...
aws_elastic_beanstalk_environment.named_test_resource: Creating...
aws_route_table_association.routeassociation1: Creation complete after 2s [id=rtbassoc-0cd7cb75ab5ce3af2]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [10s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [20s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [30s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [40s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [50s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [1m0s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [1m10s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [1m20s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [1m30s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [1m40s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [1m50s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [2m0s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [2m10s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [2m20s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [2m30s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Still creating... [2m40s elapsed]
aws_elastic_beanstalk_environment.named_test_resource: Creation complete after 2m44s [id=e-qm4jhpwpcp]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Warning: Interpolation-only expressions are deprecated

  on variables.tf line 84, in resource "aws_route_table" "route":
  84:   vpc_id = "${aws_vpc.my_vpc.id}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

(and 7 more similar warnings elsewhere)


Apply complete! Resources: 11 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:elasticbeanstalk:us-east-1:986325076436:environment/turbottest67900/turbottest67900
resource_id = e-qm4jhpwpcp
resource_name = turbottest67900

Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "arn:aws:elasticbeanstalk:us-east-1:986325076436:environment/turbottest67900/turbottest67900"
    ],
    "application_name": "turbottest67900",
    "environment_id": "e-qm4jhpwpcp",
    "environment_name": "turbottest67900",
    "partition": "aws",
    "region": "us-east-1",
    "tags": [
      {
        "Key": "elasticbeanstalk:environment-name",
        "Value": "turbottest67900"
      },
      {
        "Key": "elasticbeanstalk:environment-id",
        "Value": "e-qm4jhpwpcp"
      },
      {
        "Key": "Name",
        "Value": "turbottest67900"
      }
    ],
    "title": "turbottest67900"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:elasticbeanstalk:us-east-1:986325076436:environment/turbottest67900/turbottest67900"
    ],
    "environment_name": "turbottest67900",
    "tags": [
      {
        "Key": "elasticbeanstalk:environment-name",
        "Value": "turbottest67900"
      },
      {
        "Key": "elasticbeanstalk:environment-id",
        "Value": "e-qm4jhpwpcp"
      },
      {
        "Key": "Name",
        "Value": "turbottest67900"
      }
    ],
    "title": "turbottest67900"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "application_name": "turbottest67900",
    "environment_id": "e-qm4jhpwpcp",
    "environment_name": "turbottest67900",
    "partition": "aws",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:elasticbeanstalk:us-east-1:986325076436:environment/turbottest67900/turbottest67900"
    ],
    "region": "us-east-1",
    "title": "turbottest67900"
  }
]
✔ PASSED

POSTTEST: tests/aws_elastic_beanstalk_environment

TEARDOWN: tests/aws_elastic_beanstalk_environment

SUMMARY:

1/1 passed.

```
`### Basic ElasticBeanstalk Environment info`

Screenshot 2021-03-15 at 10 00 05 PM

`### list of Environments whose AbortableOperationInProgress is set enable`
+------------------+---------------------------------+
| environment_name | abortable_operation_in_progress |
+------------------+---------------------------------+
+------------------+---------------------------------+
(Note: Since AbortableOperationInProgress is false for all the Environments present in Account )
```
select
  environment_name,

from
  aws_elastic_beanstalk_environment
where
  abortable_operation_in_progress = 'false';
```
+------------------+---------------------------------+
| environment_name | abortable_operation_in_progress |
+------------------+---------------------------------+
| Turbot-env       | false                           |
| Rups-env-007     | false                           |
+------------------+---------------------------------+

`### list the Environments whose health is not responsive`
+------------------+----------------+--------+
| application_name | environment_id | health |
+------------------+----------------+--------+
+------------------+----------------+--------+
(Note: It's showing Empty As there is no Environments whose heath is 'Red')
```
select
  environment_name
  application_name,
  environment_id,
  health
from
  aws_elastic_beanstalk_environment
where
  health = 'Grey';
```
+------------------+----------------+--------+
| application_name | environment_id | health |
+------------------+----------------+--------+
| Turbot-env       | e-daai6sq2mc   | Grey   |
| Rups-env-007     | e-xnpckaqsgp   | Grey   |
+------------------+----------------+--------+

`### list of applications running in environment whose health status disabled`
+------------------+---------------+
| environment_name | health_status |
+------------------+---------------+
| Rups-env-007     | Suspended     |
| Turbot-env       | Suspended     |
+------------------+---------------+

### CODE REVIEW

File: aws-test/tests/aws_elastic_beanstalk_environment/variables.tf [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    }
+    //If you do not add this rule, you can not reach the NGIX
+    ingress {
+        from_port = 80
+        to_port = 80
+        protocol = "tcp"
+        cidr_blocks = ["0.0.0.0/0"]
+    }
+}
+
```
Comment (Paulami30) [inline aws-test/tests/aws_elastic_beanstalk_environment/variables.tf:122]: please use a empty line between each resource

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags assigned to the Repository",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getElasticBeanstalkEnvironmentTags,
+				Transform:   transform.FromField("ResourceTags"),
+			},
+
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:142]: please arrange the columns according to the standard column structure

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "description",
+				Description: "Describes this environment.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "endpoint_url",
+				Description: "the URL to the LoadBalancer.",
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:65]:
```suggestion
				Description: "The URL to the LoadBalancer.",
```

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+	return &plugin.Table{
+		Name:        "aws_elastic_beanstalk_environment",
+		Description: "AWS ElasticBeanstalk Environment",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.SingleColumn("environment_name"),
+			ShouldIgnoreError: isNotFoundError([]string{"ResourceNotFoundException"}),
+			Hydrate:           getAwsElasticBeanstalkEnvironment,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listAwsElasticBeanstalkEnvironment,
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:23]:
```suggestion
			Hydrate: listAwsElasticBeanstalkEnvironments,
```

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+	}
+
+	if environmentData != nil && len(environmentData.Environments) > 0 {
+		return environmentData.Environments[0], nil
+	}
+
+	return nil, nil
+}
+
+func getElasticBeanstalkEnvironmentTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:251]:
```suggestion
func listElasticBeanstalkEnvironmentTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+
+	// Build param
+	params := &elasticbeanstalk.ListTagsForResourceInput{
+		ResourceArn: resourceArn,
+	}
+
+	// Get call
+	op, err := svc.ListTagsForResource(params)
+	if err != nil {
+		logger.Debug("getElasticBeanstalkEnvironmentTags", "ERROR", err)
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:277]:
```suggestion
		logger.Debug("listElasticBeanstalkEnvironmentTags", "ERROR", err)
```

File: docs/tables/aws_elastic_beanstalk_environment.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  environment_id,
+  environment_name,
+  environment_arn,
+  tier
+from
+  aws_elastic_beanstalk_environment;
+```
+
+
+### list of Environments whose AbortableOperationInProgress is set enable
```
Comment (Paulami30) [inline docs/tables/aws_elastic_beanstalk_environment.md:21]: Is this query important?

File: docs/tables/aws_elastic_beanstalk_environment.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+select
+  environment_name
+from
+  aws_elastic_beanstalk_environment
+where
+  abortable_operation_in_progress = 'true';
+```
+
+
+### list the Environments whose health is not responsive
```
Comment (Paulami30) [inline docs/tables/aws_elastic_beanstalk_environment.md:33]:
```suggestion
### List environments whose health is not responsive
```

File: docs/tables/aws_elastic_beanstalk_environment.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+```sql
+select
+  environment_name
```
Comment (Paulami30) [inline docs/tables/aws_elastic_beanstalk_environment.md:37]: application_name,
  environment_id,

Comment (Paulami30) [inline docs/tables/aws_elastic_beanstalk_environment.md:37]:
```suggestion
  environment_name
   application_name,
   environment_id,
```

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("EnvironmentName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("ResourceTags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getElasticBeanstalkEnvironmentTags,
+				Transform:   transform.FromField("ResourceTags"),
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:155]: fetch `tags` column from elasticBeanstalkEnvironmentTagListToTurbotTags

**Review by Paulami30 — COMMENTED** [review]
Run the integration test after making the changes

File: aws/table_aws_elastic_beanstalk_environment.go (hunk 10 above)
Comment (Priyanka585464) [reply aws/table_aws_elastic_beanstalk_environment.go:155]: No, i guess this is correct.  tags should be fetch from "getElasticBeanstalkEnvironmentTags".

**Comment by Priyanka585464** [comment]
> Run the integration test after making the changes

> I have run the test again. It's working fine.

**Review by Paulami30 — COMMENTED** [review]
modify the queries as per the guideline please

File: aws/plugin.go [hunk 11]
```diff
@@ -68,6 +68,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_ec2_transit_gateway":                tableAwsEc2TransitGateway(ctx),
 			"aws_ec2_transit_gateway_route_table":    tableAwsEc2TransitGatewayRouteTable(ctx),
 			"aws_ec2_transit_gateway_vpc_attachment": tableAwsEc2TransitGatewayVpcAttachment(ctx),
+			"aws_elastic_beanstalk_environment":      tableAwsElasticBeanstalkEnvironment(ctx),
```
Comment (khushboo9024) [inline aws/plugin.go:71]: Sort the list

File: aws/service.go [hunk 12]
```diff
@@ -248,6 +258,8 @@ func EcsService(ctx context.Context, d *plugin.QueryData, region string) (*ecs.E
 	if err != nil {
 		return nil, err
 	}
+	svc := elasticbeanstalk.New(sess)
```
Comment (khushboo9024) [inline aws/service.go:261]: conflict was not resolved properly ,please check before pushing.

File: aws/table_aws_elastic_beanstalk_environment.go (hunk 2 above)
Comment (khushboo9024) [reply aws/table_aws_elastic_beanstalk_environment.go:142]: Ordering of columns is still not correct.

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+				Transform:   transform.FromField("EnvironmentArn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsElasticBeanstalkEnvironments(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_elastic_beanstalk_environment.go:170]:
```suggestion
```

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+	}
+	return op, nil
+}
+
+//// TRANSFORM FUNCTIONS
+func elasticBeanstalkEnvironmentTagListToTurbotTags(ctx context.Context, d *transform.TransformData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("elasticBeanstalkEnvironmentTagListToTurbotTags")
+	tags := h.Item.(*elasticbeanstalk.ListTagsForResourceOutput)
+
+	// Mapping the resource tags inside turbotTags
```
Comment (khushboo9024) [inline aws/table_aws_elastic_beanstalk_environment.go:288]:
```suggestion
		if tags. ResourceTags == nil {
		return nil, nil
	}
```

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+	var turbotTagsMap map[string]string
+	if tags != nil {
```
Comment (khushboo9024) [inline aws/table_aws_elastic_beanstalk_environment.go:290]:
```suggestion
	if tags. ResourceTags != nil {
```

File: aws/table_aws_elastic_beanstalk_environment.go (hunk 10 above)
Comment (khushboo9024) [reply aws/table_aws_elastic_beanstalk_environment.go:155]:
```suggestion
				Transform:   transform.FromField("ResourceTags").Transform(elasticBeanstalkEnvironmentTagListToTurbotTags),
```

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elasticbeanstalk"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
```
Comment (khushboo9024) [inline aws/table_aws_elastic_beanstalk_environment.go:12]:
```suggestion
//// TABLE DEFINITION
```

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 17]
```diff
@@ -0,0 +1,299 @@
...
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsElasticBeanstalkEnvironment,
+			},
+			{
+				Name:        "abortable_operation_in_progress",
+				Description: "Indicates if there is an in-progress environment configuration update or application version deployment that you can cancel.",
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "environment_arn",
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:77]:
```suggestion
				Name:        "arn",
```
Also make this as the 3rd column

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 18]
```diff
@@ -0,0 +1,299 @@
...
+				Name:        "endpoint_url",
+				Description: "The URL to the LoadBalancer.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsElasticBeanstalkEnvironment,
+			},
+			{
+				Name:        "abortable_operation_in_progress",
+				Description: "Indicates if there is an in-progress environment configuration update or application version deployment that you can cancel.",
+				Type:        proto.ColumnType_BOOL,
+			},
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:75]: leaving the first 5 columns, please rearrange the columns in alphabetical order

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 19]
```diff
@@ -0,0 +1,299 @@
...
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsElasticBeanstalkEnvironment,
+			},
+			{
+				Name:        "application_name",
+				Description: "The name of the application associated with this environment.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "description",
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:61]: This should be the fourth column

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 20]
```diff
@@ -0,0 +1,299 @@
...
+				Hydrate:     getAwsElasticBeanstalkEnvironment,
+			},
+			{
+				Name:        "abortable_operation_in_progress",
+				Description: "Indicates if there is an in-progress environment configuration update or application version deployment that you can cancel.",
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "environment_arn",
+				Description: "The environment's Amazon Resource Name (ARN), which can be used in other API requests that require an ARN.",
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:78]:
```suggestion
				Description: "The environment's Amazon Resource Name (ARN).",
```

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 21]
```diff
@@ -0,0 +1,299 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "environment_links",
+				Description: "A list of links to other environments in the same group.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsElasticBeanstalkEnvironment,
+			},
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:86]: All the json columns should be at the end arranged in alphabetical order. please check the standard column structure

File: aws/table_aws_elastic_beanstalk_environment.go (hunk 10 above)
Comment (Paulami30) [reply aws/table_aws_elastic_beanstalk_environment.go:157]: this change is not made yet

**Review by Paulami30 — COMMENTED** [review]
Please make the changes when you have time. Thanks

File: docs/tables/aws_elastic_beanstalk_environment.md [hunk 22]
```diff
@@ -0,0 +1,56 @@
+# Table: aws_elastic_beanstalk_environment
+
+AWS Elastic Beanstalk makes easy for you to create, deploy, and manage scalable, fault-tolerant applications running on the Amazon Web Services cloud.
```
Comment (Paulami30) [inline docs/tables/aws_elastic_beanstalk_environment.md:3]:
```suggestion
AWS Elastic Beanstalk helps to create, deploy, and manage scalable, fault-tolerant applications running on the Amazon Web Services cloud.
```

File: docs/tables/aws_elastic_beanstalk_environment.md [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+  environment_name,
+  application_name,
+  environment_id,
+  health
+from
+  aws_elastic_beanstalk_environment
+where
+  health = 'Red';
+```
+
```
[the remark this answers is not in the document]
Comment (Paulami30) [reply docs/tables/aws_elastic_beanstalk_environment.md:45]: fix the spacing for the remaining queries

File: aws/table_aws_elastic_beanstalk_environment.go [hunk 24]
```diff
@@ -0,0 +1,301 @@
...
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listElasticBeanstalkEnvironmentTags,
+				Transform:   transform.FromField("ResourceTags"),
+			},
+			{
+				Name:        "tier",
+				Description: "Describes the current tier of this environment.",
+				Type:        proto.ColumnType_JSON,
+			},
+
```
Comment (Paulami30) [inline aws/table_aws_elastic_beanstalk_environment.go:144]: {
				Name:        "tier",
				Description: "Describes the current tier of this environment.",
				Type:        proto.ColumnType_JSON,
			},
			{
				Name:        "tags_src",
				Description: "A list of tags assigned to the Repository",
				Type:        proto.ColumnType_JSON,
				Hydrate:     listElasticBeanstalkEnvironmentTags,
				Transform:   transform.FromField("ResourceTags"),
			},

Comment (Paulami30) [reply aws/table_aws_elastic_beanstalk_environment.go:144]: we usually keep the tags_src as the last column(but before the standard columns)

File: docs/tables/aws_elastic_beanstalk_environment.md [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+  environment_id,
+  environment_name,
+  application_name,
+  environment_arn,
+  tier
+from
+  aws_elastic_beanstalk_environment;
+```
+
+### List environments whose configuration update enable
```
Comment (Paulami30) [inline docs/tables/aws_elastic_beanstalk_environment.md:20]:
```suggestion
### List environments which has configuration update or application version deployment in-progress 
```

File: docs/tables/aws_elastic_beanstalk_environment.md [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+
+```sql
+select
+  environment_name,
+  abortable_operation_in_progress
+from
+  aws_elastic_beanstalk_environment
+where
+  abortable_operation_in_progress = 'true';
+```
```
[the remark this answers is not in the document]
Comment (Priyanka585464) [reply docs/tables/aws_elastic_beanstalk_environment.md:30]: Yes, i think so. Client can get the idea if there is an in-progress environment configuration update or application version deployment that you can cancel.

**Review by Paulami30 — APPROVED** [review]

**Comment by Paulami30** [comment]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM now

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**